src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -191,7 +192,17 @@ private ImmutableArray<ISourceFile> CloseCompilationInternal(DocumentUri documen
                             }
                         }
 
-                        return this.provider.Create(workspace, documentUri, modelLookup.ToImmutableDictionary());
+                        ConfigHelper? configHelper = null;
+ 
+                        if (!reloadBicepConfig &&
+                        activeContexts.TryGetValue(documentUri, out CompilationContext? compilationContext) &&
+                        compilationContext is not null &&
+                        compilationContext.Compilation.ConfigHelper is ConfigHelper previousConfigHelper)
+                        {
+                            configHelper = previousConfigHelper;
+                        }
```

Comment:
Can this be simplified to:

```csharp
var configHelper = activeContexts.TryGetValue(documentUri)?.Compilation.ConfigHelper;
if (reloadBicepConfig)
{
    configHelper = null;
}
```

Reply:
EDIT: In fact, looks like you already have access to the previous context with the `prevContext` variable, so you shouldn't need to fetch it back with `TryGetValue()`

Reply:
Simplified as part of this commit:
https://github.com/Azure/bicep/commit/74db298612aac1a8c0d155fca8b08251f23cf1ae